sretoolbox/container/image.py

```diff
@@ -353,9 +382,17 @@ def _request_get(self, url):
 
             # Try again, this time with the Authorization header
             headers['Authorization'] = self._get_auth(www_auth)
-            response = requests.get(url, headers=headers)
+            response = requests.head(url, headers=headers)
+
+        if 'etag' in response.headers:
+            with contextlib.suppress(KeyError):
+                return self.response_cache[response.headers['etag']]
 
+        response = requests.get(url, headers=headers, auth=self.auth)
         self._raise_for_status(response)
+        if self._should_cache(response):
```

Comment:
we did a lot before even knowing if we should cache. This should probably be the first thing to check before hitting the cache backend or doing a HEAD request.